### PR TITLE
Add basic arithmetic functions to calculator2.py

### PR DESCRIPTION
**Pull request by Slyracoon23**
Type of PR:
**Enhancement**

___
PR Description:
**This PR introduces four basic arithmetic functions (addition, subtraction, multiplication, and division) to the calculator2.py file. It also includes a check in the division function to prevent division by zero.**

___
PR Main Files Walkthrough:
- `calculator2.py`: Added functions for addition, subtraction, multiplication, and division. Included a check in the division function to prevent division by zero.

### CODE REVIEW

**Comment by Slyracoon23** [comment]
Preparing pr unittest...

**Comment by Slyracoon23** [comment]
Preparing pr unittest...

**Comment by Slyracoon23** [comment]
Preparing pr unittest...

**Comment by Slyracoon23** [comment]
Create calculator2.py
Type of PR:
**Other**

___
PR Description:
**This PR introduces a new file named 'calculator2.py'. The file contains four basic mathematical functions: addition, subtraction, multiplication, and division. Each function takes two parameters, performs the respective operation, and returns the result. Notably, the division function includes a check to prevent division by zero, raising a ValueError if such an operation is attempted.**

___
PR Unit Tests Description:
**No unit tests are included in this PR. However, for each function, tests should be created to verify correct operation under normal conditions, as well as edge cases. For example, the 'divide' function should be tested with a denominator of zero to ensure the ValueError is properly raised.**

___
PR Unit Tests Code:
**```python
import unittest
import calculator2

class TestCalculator(unittest.TestCase):
    def test_add(self):
        self.assertEqual(calculator2.add(5, 3), 8)

    def test_subtract(self):
        self.assertEqual(calculator2.subtract(5, 3), 2)

    def test_multiply(self):
        self.assertEqual(calculator2.multiply(5, 3), 15)

    def test_divide(self):
        self.assertEqual(calculator2.divide(6, 3), 2)
        with self.assertRaises(ValueError):
            calculator2.divide(5, 0)

if __name__ == '__main__':
    unittest.main()
```**

___

**Comment by Slyracoon23** [comment]
Preparing pr description...

**Comment by Slyracoon23** [comment]
Preparing pr description...

**Comment by Slyracoon23** [comment]
Preparing review...

**Comment by Slyracoon23** [comment]
## PR Analysis

- 🎯 **Main theme:** Adding basic arithmetic functions to calculator2.py
- 📌 **Type of PR:** Enhancement
- 🧪 **Relevant tests added:** No
- ✨ **Focused PR:** Yes, because the title and description clearly state that the PR is about adding basic arithmetic functions to calculator2.py, and all changes in the PR are related to this task.
- 🔒 **Security concerns:** No, because the PR code does not involve any operations that could potentially lead to security issues such as SQL injection, XSS, CSRF, etc.
## PR Feedback

- 💡 **General PR suggestions:** The PR is well-structured and focused. However, it lacks tests for the new functions. It's important to add tests to ensure the functions work as expected. Also, consider adding type hints to the function signatures for better readability and to prevent potential type-related bugs.

File: calculator2.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+def add(x, y):
```
Comment (Slyracoon23) [inline calculator2.py:1]: Add type hints to function signatures. For example, the addition function can be defined as `def add(x: float, y: float) -> float:`. This will make the code more readable and help prevent potential type-related bugs. [important]

File: calculator2.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+def subtract(x, y):
+    return x - y
+
+def multiply(x, y):
+    return x * y
+
+def divide(x, y):
+    if y == 0:
+        raise ValueError("Cannot divide by zero!")
```
Comment (Slyracoon23) [inline calculator2.py:12]: Consider handling the case when the user tries to divide by zero more gracefully. Instead of raising a ValueError, you could return a specific message or a special value like 'inf'. [medium]
